app/src/main/java/govind/iiitl/app/activities/VerifyActivity.kt

```diff
@@ -17,6 +18,16 @@ class VerifyActivity : AppCompatActivity() {
         submitBtn.setOnClickListener {
             submit()
         }
+
+        verifyText.setOnEditorActionListener { v, actionId, event ->
+             when (actionId) {
+                EditorInfo.IME_ACTION_DONE -> {
+                    submit()
```

Comment:
Do not call submit here. Add this to close keyboard on pressing done
`                    val inputMethodManager: InputMethodManager = getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
                    inputMethodManager.hideSoftInputFromWindow(v.applicationWindowToken, 0)`